leftwm-core/src/utils/state_socket.rs

```diff
@@ -61,7 +61,9 @@ impl StateSocket {
             let mut json = serde_json::to_string(&state)?;
             json.push('\n');
             let mut state = self.state.lock().await;
-            if json != state.last_state {
+
+            let state_changed = json != state.last_state;
+            if state_changed {
```

Comment:
I think that's good for code readability. My guess is the compiler just optimizes that variable away anyway right?

Reply:
> My guess is the compiler just optimizes that variable away anyway right?

yes it does!